analyzers/src/SonarAnalyzer.CSharp/Extensions/SyntaxNodeAnalysisContextExtensions.cs

```diff
@@ -0,0 +1,32 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2021 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using StyleCop.Analyzers.Lightup;
+
+namespace SonarAnalyzer.Extensions
+{
+    public static class SyntaxNodeAnalysisContextExtensions
+    {
+        public static bool IsTopLevelCompilationUnit(this SyntaxNodeAnalysisContext context) =>
+           context.ContainingSymbol is IMethodSymbol { Name: INamedTypeSymbolExtensions.MainMethodImplicitName };
```

Comment:
I think it would be better if there was a new declaration of a `private const MainMethodImplicitName` rather than turning into `public` the one from `INamedTypeSymbolExtensions`.
In case we decide to change something in the other class or remove the function from there for example, we'll need to do a change also here.

Reply:
There should be as less duplicates as possible. @costin-zaharia-sonarsource agreed to find a better place for this constant in the near future.